### PR TITLE
Trying to make use of loop device more flexible.

### PR DESCRIPTION
**Pull request by rhauns**
Main changes are in config-ubuntu.sh:
* grep for the MP in the list of the mounted file systems and extract the loop device
* match: the VM is already mounted and uses the extracted loop device
* no match: get the next unused loop device using _losetup -f_

Changes in the other scripts prevent actions which require mount/unmount state.

Drawback:
There is still a "hardcoded" _loop0_ in the grub-installation scripts. So installation fails if any other loop device as 0 is taken.

### CODE REVIEW

File: rootfs/root/bin/config-ubuntu.sh [hunk 1]
```diff
@@ -14,8 +14,17 @@ LE_SIZE=4
...
 MP=/tmp/mp-${VM_NAME}
+MOUNTED=`df -h | grep ${MP} | grep -oP "loop\d*"`
+
+if [ -z ${MOUNTED} ]; then
+	# The mount point has actually no loop device mounted, 
+	# take the next free loop device.
+	LOOP=`losetup -f | cut -d/ -f3`
+	SWAP=/dev/mapper/${LOOP}p2
+	EXT4=/dev/mapper/${LOOP}p3
+else
```
Comment (stmork) [inline rootfs/root/bin/config-ubuntu.sh:25]: I would suggest to define vars SWAP and EXT4 also in the else part.

File: rootfs/root/bin/config-ubuntu.sh [hunk 2]
```diff
@@ -14,8 +14,17 @@ LE_SIZE=4
 LV_NAME=${VM_NAME}-disk01
 DISK=/dev/${VG_NAME}/${LV_NAME}
 MP=/tmp/mp-${VM_NAME}
+MOUNTED=`df -h | grep ${MP} | grep -oP "loop\d*"`
+
```
Comment (stmork) [inline rootfs/root/bin/config-ubuntu.sh:18]: Think about the _mountpoint_ tool if $MP is already mounted. It simplifies a lot.

**Review by stmork — CHANGES_REQUESTED** [review]

File: rootfs/root/bin/config-ubuntu.sh (hunk 1 above)
Comment (rhauns) [reply rootfs/root/bin/config-ubuntu.sh:25]: Reflected in this [commit](https://github.com/stmork/kvm-hostcreator/commit/cf65187391f1d0f229a74a0bc94856da74b29807).

File: rootfs/root/bin/config-ubuntu.sh (hunk 2 above)
Comment (rhauns) [reply rootfs/root/bin/config-ubuntu.sh:18]: Working on it [here](https://github.com/stmork/kvm-hostcreator/tree/rhauns-dev).

**Comment by rhauns** [comment]
Closing this PR from my fork, go on working in a dev-branch of Steffens repo an re-opening a new PR from there.

**Pull request closed without merging**